### PR TITLE
Fix overflow of 32-bit integer during compaction

### PR DESCRIPTION
**Pull request by vladimir-bukhtoyarov**
This is fix for https://issues.apache.org/jira/browse/CASSANDRA-14773

### CODE REVIEW

File: src/java/org/apache/cassandra/utils/streamhist/StreamingTombstoneHistogramBuilder.java [hunk 1]
```diff
@@ -595,12 +600,22 @@ private boolean tryCell(int cell, int point, int delta)
...
-    private static int roundKey(int p, int roundSeconds)
+    private static int roundKey(int point, int roundSeconds)
     {
-        int d = p % roundSeconds;
+        int d = point % roundSeconds;
         if (d > 0)
-            return p + (roundSeconds - d);
+        {
+            point += roundSeconds - d;
+            if (point == Cell.MAX_DELETION_TIME)
```
Comment (belliottsmith) [inline src/java/org/apache/cassandra/utils/streamhist/StreamingTombstoneHistogramBuilder.java]: This should be >

File: src/java/org/apache/cassandra/utils/streamhist/StreamingTombstoneHistogramBuilder.java [hunk 2]
```diff
@@ -595,12 +600,22 @@ private boolean tryCell(int cell, int point, int delta)
...
+                // the rounding should not be a reason to mark point as unexpirable
+                point =  Cell.MAX_DELETION_TIME;
```
Comment (belliottsmith) [inline src/java/org/apache/cassandra/utils/streamhist/StreamingTombstoneHistogramBuilder.java]: why assign `point =` here, but `return` below?

File: src/java/org/apache/cassandra/utils/streamhist/StreamingTombstoneHistogramBuilder.java (hunk 1 above)
Comment (vladimir-bukhtoyarov) [reply src/java/org/apache/cassandra/utils/streamhist/StreamingTombstoneHistogramBuilder.java]: Fixed. The new testcase "testThatPointIsNotMissedBecauseOfRoundingToNoDeletionTime" added to cover this branch

File: src/java/org/apache/cassandra/utils/streamhist/StreamingTombstoneHistogramBuilder.java (hunk 2 above)
Comment (vladimir-bukhtoyarov) [reply src/java/org/apache/cassandra/utils/streamhist/StreamingTombstoneHistogramBuilder.java]: Fast return applied

**Comment by vladimir-bukhtoyarov** [comment]
I am closing this, because of a similar pull-request https://github.com/apache/cassandra/pull/491 that has been accepted.

**Pull request closed without merging**